File: android/app/src/main/java/com/preuvely/app/data/models/Report.kt
package com.preuvely.app.data.models

import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*

enum class ReportableType(val value: String) {
    @SerializedName("store") STORE("store"),
    @SerializedName("review") REVIEW("review"),
    @SerializedName("reply") REPLY("reply")
}

enum class ReportReason(val value: String) {
    @SerializedName("spam") SPAM("spam"),
    @SerializedName("abuse") ABUSE("abuse"),
    @SerializedName("fake") FAKE("fake"),
    @SerializedName("inappropriate") INAPPROPRIATE("inappropriate"),
    @SerializedName("other") OTHER("other");

    val displayName: String
        get() = when (this) {
            SPAM -> "Spam"
            ABUSE -> "Abusive Content"
            FAKE -> "Fake Store/Review"
            INAPPROPRIATE -> "Inappropriate Content"
            OTHER -> "Other Issue"
        }

    val description: String
        get() = when (this) {
            SPAM -> "This content is promotional or spam"
            ABUSE -> "Contains harassment or hate speech"
            FAKE -> "Fake or misleading information"
            INAPPROPRIATE -> "Contains adult or offensive content"
            OTHER -> "Other issue not listed above"
        }

    val iconName: String
        get() = when (this) {
            SPAM -> "exclamationmark_triangle"
            ABUSE -> "hand_raised"
            FAKE -> "doc_badge_ellipsis"
            INAPPROPRIATE -> "eye_slash"
            OTHER -> "questionmark_circle"
        }
}

enum class ReportStatus(val value: String) {
    @SerializedName("open") OPEN("open"),
    @SerializedName("under_review") UNDER_REVIEW("under_review"),
    @SerializedName("resolved") RESOLVED("resolved"),
    @SerializedName("dismissed") DISMISSED("dismissed")
}

data class Report(
    @SerializedName("id") val id: Int,
    @SerializedName("reportable_type") val reportableType: String,
    @SerializedName("reportable_id") val reportableId: Int,
    @SerializedName("reportable_name") val reportableName: String?,
    @SerializedName("reason") val reason: ReportReason,
    @SerializedName("note") val note: String?,
    @SerializedName("status") val status: ReportStatus,
    @SerializedName("created_at") val createdAt: String?
) {
    val formattedDate: String
        get() {
            createdAt ?: return ""
            return try {
                val inputFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
                val outputFormat = SimpleDateFormat("MMM d, yyyy", Locale.getDefault())
                val date = inputFormat.parse(createdAt)
                date?.let { outputFormat.format(it) } ?: ""
            } catch (e: Exception) {
                ""
            }
        }

    val displayableType: String
        get() = when (reportableType) {
            "store" -> "Store"
            "review" -> "Review"
            "reply" -> "Reply"
            else -> reportableType.replaceFirstChar { it.uppercase() }
        }

    val displayReportableName: String
        get() = reportableName ?: "Unknown"
}

data class ReportListResponse(
    @SerializedName("data") val data: List<Report>
)

data class ReportResponse(
    @SerializedName("message") val message: String,
    @SerializedName("data") val data: Report
)

File: ios/Preuvely/Preuvely/Models/Report.swift
import Foundation

// MARK: - Reportable Type

enum ReportableType: String, Codable {
    case store
    case review
    case reply
}

// MARK: - Report Reason

enum ReportReason: String, Codable, CaseIterable, Identifiable, Hashable {
    case spam
    case abuse
    case fake
    case other

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .spam: return "Spam"
        case .abuse: return "Abuse"
        case .fake: return "Fake/Misleading"
        case .other: return "Other"
        }
    }

    var localizedDisplayName: String {
        switch self {
        case .spam: return L10n.Report.Reason.spam.localized
        case .abuse: return L10n.Report.Reason.abuse.localized
        case .fake: return L10n.Report.Reason.fake.localized
        case .other: return L10n.Report.Reason.other.localized
        }
    }

    var iconName: String {
        switch self {
        case .spam: return "exclamationmark.bubble.fill"
        case .abuse: return "hand.raised.fill"
        case .fake: return "eye.slash.fill"
        case .other: return "ellipsis.circle.fill"
        }
    }
}

// MARK: - Report Status

enum ReportStatus: String, Codable, Hashable {
    case open
    case resolved
    case dismissed

    var displayName: String {
        switch self {
        case .open: return "Open"
        case .resolved: return "Resolved"
        case .dismissed: return "Dismissed"
        }
    }
}

// MARK: - Report

struct Report: Identifiable, Codable, Hashable {
    let id: Int
    let reportableType: String // Backend returns lowercase type: "store", "review", "reply"
    let reportableId: Int
    let reportableName: String?
    let reason: ReportReason
    let note: String?
    let status: ReportStatus
    let createdAt: Date?

    // Note: Using JSONDecoder with .convertFromSnakeCase strategy

    var formattedDate: String {
        guard let date = createdAt else { return "" }
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .none
        return formatter.string(from: date)
    }

    /// Get the reportable type for display (converts backend type to user-friendly text)
    var displayableType: String {
        switch reportableType.lowercased() {
        case "store": return "Store"
        case "review": return "Review"
        case "storereply", "reply": return "Reply"
        default: return reportableType.capitalized
        }
    }

    /// Display name for the reported content (falls back to type + ID if name not available)
    var displayReportableName: String {
        reportableName ?? "\(displayableType) #\(reportableId)"
    }
}

// MARK: - Preview Data

#if DEBUG
extension Report {
    static let samples: [Report] = []
    static let sample = Report(
        id: 1,
        reportableType: "store",
        reportableId: 1,
        reportableName: "Preview Store",
        reason: .other,
        note: nil,
        status: .open,
        createdAt: Date()
    )
}
#endif
